src/account/TransferTransaction.js

```diff
@@ -244,6 +249,24 @@ export default class TransferTransaction extends Transaction {
         return this;
     }
 
+    /**
+     * @internal
+     * @param {AccountId | string} accountId
+     * @param {number | string | Long | LongObject | BigNumber | Hbar} amount
+     * @returns {number | Long}
+     */
+    squashTransfers(accountId, amount) {
+        let currentValue = this.hbarTransfers.get(accountId);
+        return Number(
+            Hbar.fromTinybars(
+                (currentValue == null
+                    ? Long.ZERO
+                    : currentValue.toTinybars()
+                ).add(new Hbar(Number(amount)).toTinybars())
```

Comment:
Instead of casting `amount` to `Number` and passing that into `new Hbar()` you should do `amount instanceof Hbar ? amount : new Hbar(amount)`